src/components/UniversalAmount.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+    <span class="amount" :class="{ approx: showApprox && isApprox }">
+        {{ formattedAmount }}
+        <span class="currency" :class="currency">{{currency}}</span>
+    </span>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import bigInt from 'big-integer';
+
+@Component
+export default class UniversalAmount extends Vue {
+    /**
+     * Amount in smallest unit
+     */
+    @Prop({type: [Number, Object]}) public amount!: number | bigInt.BigInteger;
+    /**
+     * Actual decimal count of the currency being displayed
+     */
+    @Prop({type: Number, default: 5}) public decimals!: number;
+    @Prop({type: Number, default: 2}) public minDecimals!: number;
+    @Prop({type: Number, default: 5}) public maxDecimals!: number;
+    @Prop({type: Boolean, default: false}) public showApprox!: boolean;
+    @Prop({type: String, default: 'nim'}) public currency;
+
+    private isApprox: boolean = false;
+
+    private get formattedAmount() {
+        let result: string;
+
+        if (typeof this.amount === 'number') {
+            const value = Math.round(
+                    this.amount * 10 ** this.maxDecimals / 10 ** this.decimals,
+                ) / 10 ** this.maxDecimals;
```

Comment:
Should reintroduce the `roundingFactor` variable as in `Amount.vue` as that's much more readable.